### ISSUE TITLE
Support `binding_array` of Uniform Buffers on DX12

### ISSUE THREAD

**Issue by cwfitzgerald** [gfx-rs/wgpu#6740]
LIke #6739 but using `ConstantBuffer`

**Comment by 39ali** [comment]
i would like to work on it but im not sure what's the difference between `UNIFORM_BUFFER_BINDING_ARRAYS` and [BUFFER_BINDING_ARRAY](https://docs.rs/wgpu/latest/wgpu/struct.Features.html#associatedconstant.BUFFER_BINDING_ARRAY) ?
are they the same ?

**Comment by cwfitzgerald** [comment]
No, BUFFER_BINDING_ARRAYS is storage buffers and UNIFORM_BUFFER_BINDING_ARRAYS is uniform buffers

**Comment by 39ali** [comment]
@cwfitzgerald so i think the docs needs to be updated? because uniforms are being used here : 



pub const [BUFFER_BINDING_ARRAY](https://docs.rs/wgpu-types/29.0.0/wgpu_types/struct.Features.html#associatedconstant.BUFFER_BINDING_ARRAY): Self
Allows the user to create arrays of buffers in shaders:

ex.

var< uniform >  buffer_array: array<MyBuffer, 10> (WGSL)
uniform myBuffer { ... } buffer_array[10] (GLSL)
This capability allows them to exist and to be indexed by dynamically uniform values.
